collector.go

```diff
@@ -39,18 +39,20 @@ func (c *collector) collect() (flatMetrics, error) {
 
 	m, err := cl.metrics()
 	if err != nil {
-		return nil, err
+		log.Printf("failed to get metrics: %v", err)
 	}
 
 	all := transformMetrics(m)
 
 	/* optional syncing stats */
 	s, err := cl.syncingMetrics()
-	if err == nil {
-		sync := decodeSyncData(s, "sync_")
-		for k, v := range sync {
-			all[k] = v
-		}
+	if err != nil {
+		log.Printf("failed to get syncing stats: %v", err)
+	}
+
+	sync := decodeSyncData(s, "sync_")
```

Comment:
Wrong, this is precisely why I'm not returning here, because I want the endpoint to return something even if both RPC calls fail. But really, now I'm mostly interested in the `syncingMetrics`, since as you said yourself `1.9.0` has its own endpoint for Prometheus metrics, which we will start using soon, working on it.

Reply:
Ok, let's add a comment why `nil` value is fine 👍